backend/packages/Upgrade/src/api/repositories/ExperimentRepository.ts

```diff
@@ -76,10 +76,11 @@ export class ExperimentRepository extends Repository<Experiment> {
       .where(
         new Brackets((qb) => {
           qb.where(
-            '(experiment.state = :enrolling OR experiment.state = :enrollmentComplete) AND :context ILIKE ANY (ARRAY[experiment.context])',
+            '(experiment.state = :enrolling OR NOT (experiment.state = :enrollmentComplete AND experiment.postExperimentRule = :assign AND experiment.revertTo IS NULL)) AND :context ILIKE ANY (ARRAY[experiment.context])',
```

Comment:
This query will return inactive, preview and other experiments. It should be something like this
enrolling OR enrollmentComplete AND NOT (enrollmentComplete and postRule = assign and revertTo is NULL)
or 
enrolling OR (enrollmentComplete and revertTo is NOT NULL)